### ISSUE TITLE
Displaying readable errors

### ISSUE THREAD

**Issue by vincentorback** [MadLittleMods/postcss-css-variables#22]
For now it looks like this great plugin doesn’t log any errors at all.
If I for example have a setup like this:

```css
:root {
  --my-color: red;
}

.btn {
  color: var(--other-color);
}
```

All I get is a postcss error that looks something like this: 
`TypeError: Cannot read property 'indexOf' of undefined]`

Would be nice if this plugin could show what the error is, which file and which linenumber. And maybe using [postcss-reporter](https://github.com/postcss/postcss-reporter).

**Comment by MadLittleMods** [comment]
Hey @vincentorback,

I have also run into this before but I am having trouble reproducing it now even with your example.

It should be [adding to the warnings](https://github.com/MadLittleMods/postcss-css-variables/blob/master/lib/resolve-value.js#L83) but it seems to get caught up in some code and error out before that.

**Comment by vincentorback** [comment]
Okey... I tried to have a look but was hard to track down the issue.
How is it set up now? It just ignores errors and displays `undefined`, or should the plugin crash, or something else?

**Comment by MadLittleMods** [comment]
It should display `undefined` and that will invalidate the declaration in the browser to potentially fallback to other values. It will also emit a warning on the `result` object which `postcss-reporter` inspects and displays.

**Comment by vincentorback** [comment]
I’m using the 0.5.1 version. And also in running PostCSS with `npm script`.

After lots of testing with and without other plugins and moving around files it boiled down it not working with autoprefixer: 
```bash
postcss -u postcss-css-variables -u autoprefixer -u postcss-reporter ./style.css
```
which is really weird but perhaps it picks up error differently and does something new? I’m going to keep looking for what’s wrong but this is just my last finding.

**Comment by vincentorback** [comment]
An idea which I'm not sure is true, but could it be so that `postcss-css-variables` sets the property to `undefined` and not `"undefined"`, a string, so that when autoprefixer picks up the trail it fails?

**Comment by tbremer** [comment]
I am having the same issue here, it's not a big deal to move `autoprefixer` to a second build, _however_ I am not seeing any warnings or reports generated when an incorrect variable declaration is being passed in.

Thoughts?

versions:
```
"postcss": "^5.0.14"
"postcss-css-variables": "^0.5.1"
```

**Comment by MadLittleMods** [comment]
@tbremer Create a separate issue as this seems unrelated.

**Comment by tbremer** [comment]
@MadLittleMods, sounds good. I will try to get a testable situation written up as well!

**Comment by vincentorback** [comment]
Still not sure if this is a `postcss-css-variables`-issue but I did some more tests:

```css
.foo {
  color: var(--bar);
}
```

`-u postcss-css-variables style.css` Works fine, rendering `.foo {color:undefined}`.

`-u postcss-css-variables -u autoprefixer style.css` Breaks showing error.

`-u postcss-css-variables -u cssnano style.css` Breaks showing error.

So the issue is when other plugins reads the result of a missing variables which is `undefined`. But if you just write:
```css
.blue {
  color: undefined
}
```
It works fine... Not trying to rush a fix, I don’t think it’s a huge problem, but just to let you know.

**Comment by MadLittleMods** [comment]
No worries @vincentorback, I really appreciate the barebones test samples and results!

If the `undefined` value is messing things up, we can probably just check and convert it to a string either [here](https://github.com/MadLittleMods/postcss-css-variables/blob/4265ae711f5e5832516e3205daba925d8dfb8b03/lib/resolve-decl.js#L98) or [here](https://github.com/MadLittleMods/postcss-css-variables/blob/4265ae711f5e5832516e3205daba925d8dfb8b03/lib/resolve-decl.js#L145)

**Comment by MadLittleMods** [comment]
Released in [v0.6.0](https://github.com/MadLittleMods/postcss-css-variables/blob/master/CHANGELOG.md#v060---2016-9-23)  :tada:. Thanks to @vincentorback for the contribution ❤️
